projects/portal/src/app/pages/dialogs/delegate/redelegate-form-dialog/redelegate-form-dialog.component.ts

```diff
@@ -81,12 +88,34 @@ export class RedelegateFormDialogComponent implements OnInit {
   ngOnInit(): void {}
 
   async onSubmit($event: RedelegateOnSubmitEvent) {
-    const txHash = await this.stakingAppService.Redelegate(
-      this.validator?.operator_address!,
-      $event.destinationValidator,
-      $event.amount,
-      $event.minimumGasPrice,
-    );
-    this.matDialogRef.close(txHash);
+    this.validatorsDetailList$.subscribe(async (valList) => {
+      const validatorStatus = valList?.find(
+        (val) => val.operator_address == $event.destinationValidator,
+      )?.status;
+      if (validatorStatus != 'BOND_STATUS_BONDED') {
+        const inactiveValidatorResult = await this.dialog
+          .open(InactiveValidatorConfirmDialogComponent, {
+            data: { valAddress: $event.destinationValidator, isConfirmed: false },
+          })
+          .afterClosed()
+          .toPromise();
+
+        if (
+          inactiveValidatorResult === undefined ||
+          inactiveValidatorResult.isConfirmed === false
+        ) {
+          this.snackBar.open('Delegate was canceled', undefined, { duration: 6000 });
+          return;
+        }
+      }
+
+      const txHash = await this.stakingAppService.Redelegate(
+        this.validator?.operator_address!,
+        $event.destinationValidator,
+        $event.amount,
+        $event.minimumGasPrice,
+      );
+      this.matDialogRef.close(txHash);
+    });
```

Comment:
@Senna46 
Sorry, I missed that.

For validatorsList$, if new data flows in, e.g. a different node is selected, is there a possibility that a transaction that has already been executed will be executed again?

I thought it might be better to add a modification such as unsubscribe after the process is complete, or convert it to a Promise.

Reply:
This part alone seems fine to `subscribe` , but I will correct it just in case.